components/TaskItem.test.tsx

```diff
@@ -0,0 +1,118 @@
+// components/TaskItem.test.tsx
+import React from 'react'
+import { describe, it, beforeEach, vi, expect } from 'vitest'
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import TaskItem from './TaskItem'
+import { Task } from '../types/task'
+
+// next/link をシンプルな <a> に置き換え
+vi.mock('next/link', () => ({
+  __esModule: true,
+  default: ({ href, children }: any) => <a href={href}>{children}</a>,
+}))
+
+// useRouter をダミー実装
+vi.mock('next/navigation', () => ({
+  useRouter: () => ({ push: vi.fn() }),
+}))
+
+describe('TaskItemのテスト', () => {
+  let mockToggle: (id: string) => void
+  let mockDelete: (id: string) => void
+  const task: Task = {
+    id: '1',
+    title: 'テストタイトル',
+    description: 'テスト説明',
+    completed: false,
+  }
+
+  beforeEach(() => {
+    mockToggle = vi.fn()
+    mockDelete = vi.fn()
+  })
```

Comment:
現在のプロジェクトで、モックをアサインしないで、リセットします
```suggestion
  beforeEach(() => {
    mockToggle.mockReset()
    mockDelete.mockReset()
  })
```